src/backend/Dimacs.ml

```diff
@@ -4,118 +4,135 @@ Copyright 2014 Guillaume Bury
+    Format.fprintf fmt "c %s@,%a@," name (Vec.pp ~sep:"" pp_dimacs) vec
 
   let export_assumption fmt vec =
-    Format.fprintf fmt "c Local assumptions@,a %a@," St.pp_dimacs vec
+    Format.fprintf fmt "c Local assumptions@,a %a@," pp_dimacs vec
 
   let export_icnf_aux r name map_filter fmt vec =
-    let aux fmt v =
+    let aux fmt _ =
       for i = !r to (Vec.size vec) - 1 do
         let x = Vec.get vec i in
         match map_filter x with
         | None -> ()
-        | Some y -> Format.fprintf fmt "%a@," St.pp_dimacs (Vec.get vec i)
+        | Some _ -> Format.fprintf fmt "%a@," pp_dimacs (Vec.get vec i)
       done;
       r := Vec.size vec
     in
     Format.fprintf fmt "c %s@,%a" name aux vec
 
+  (* TODO
   let map_filter_learnt c =
-    match c.St.cpremise with
+    match P.Clause.premise c with
     | St.Hyp | St.Local -> assert false
     | St.Lemma _ -> Some c
     | St.History l ->
       begin match l with
         | [] -> assert false
         | d :: _ ->
-          begin match d.St.cpremise with
+          begin match St.Clause.premise d with
             | St.Lemma _ -> Some d
             | St.Hyp | St.Local | St.History _ -> None
           end
       end
 
   let filter_vec learnt =
-    let lemmas = Vec.make (Vec.size learnt) St.dummy_clause in
+    let lemmas = Vec.create() in
     Vec.iter (fun c ->
         match map_filter_learnt c with
         | None -> ()
         | Some d -> Vec.push lemmas d
       ) learnt;
     lemmas
+     *)
 
-  let export fmt ~hyps ~history ~local =
-    assert (Vec.for_all (function
-        | { St.cpremise = St.Hyp; _} -> true | _ -> false
-      ) hyps);
+  let export st fmt ~hyps ~history ~local =
+    assert false
+    (* FIXME
+    assert (Vec.for_all (fun c -> St.Clause.premise c = St.Hyp) hyps);
     (* Learnt clauses, then filtered to only keep only
        the theory lemmas; all other learnt clauses should be logical
        consequences of the rest. *)
     let lemmas = filter_vec history in
+    let lemmas = history in
     (* Local assertions *)
-    assert (Vec.for_all (function
-        | { St.cpremise = St.Local; _} -> true | _ -> false
-      ) local);
+    assert (Vec.for_all (fun c -> St.Local = St.Clause.premise c) local);
     (* Number of atoms and clauses *)
-    let n = St.nb_elt () in
+    let n = St.nb_elt st in
     let m = Vec.size local + Vec.size hyps + Vec.size lemmas in
     Format.fprintf fmt
       "@[<v>p cnf %d %d@,%a%a%a@]@." n m
       (export_vec "Local assumptions") local
       (export_vec "Hypotheses") hyps
       (export_vec "Lemmas") lemmas
+       *)
 
   (* Refs to remember what portion of a problem has been printed *)
   let icnf_hyp = ref 0
   let icnf_lemmas = ref 0
 
   let export_icnf fmt ~hyps ~history ~local =
-    assert (Vec.for_all (function
-        | { St.cpremise = St.Hyp; _} -> true | _ -> false
-      ) hyps);
+    assert false
+    (* FIXME
+    assert (Vec.for_all (fun c -> St.Clause.premise c = St.Hyp) hyps);
```

Comment:
Same question. Do the changes to clauses require these functions to be changed significantly ?